homeassistant/components/light/flux_led.py

```diff
@@ -242,7 +248,7 @@ def turn_on(self, **kwargs):
             self._bulb.setPresetPattern(EFFECT_MAP[effect], 50)
 
         # white change only
-        elif white is not None:
+        if white is not None:
```

Comment:
The changes from `elif`s to `if`s will make it do four color change calls to the bulb if `rgb`, `brightness`, `effect` and `white` are all set. Even with just `rgb` and `brightness`, it'll still go through the whole "fetch rgb" block. This method should only ever call a single `setRgb`, from the looks of it.

Simply returning after each setRgb would be one way to go here.

Reply:
Wait, scratch that, as the first two ifs are mutually exclusive (I recommend switching that second block to elif to make that clearer.) It still seems like there's an issue if `brightness` and `effect` are both defined, as it'll make two `.setRgb` calls

Reply:
@OverloadUT I was able to simplify the conditionals even further by specifying fallback brightness value if argument not passed.  This allowed the use of a single call to `setRgb` as you mentioned.

Setting white value requires a separate call, so I left that as a separate conditional.  

Effects override rgb, brightness and white levels so no need to support simultaneous setting of values when an effect is used.